src/simulation/point.py

```diff
@@ -215,4 +236,13 @@ def pour_from_buffer(self):
         self.viscosity_dynamic = (sum([tup[0] * tup[1] for tup in self.oil_buffer]) + self.oil_mass * self.viscosity_dynamic) / oil_mass
         self.emulsification_rate = (sum([tup[0] * tup[2] for tup in self.oil_buffer]) + self.oil_mass * self.emulsification_rate) / oil_mass
         self.oil_buffer = []
-        self.oil_mass = oil_mass
\ No newline at end of file
+        self.oil_mass = oil_mass
+
+    def add_oil_to_other(self, coord: Coord_t, mass: float) -> None:
+        self.oil_mass -= mass
+        if not (0 <= coord[0] < const.point_side_count and 0 <= coord[1] < const.point_side_count):
+            return
+        if coord not in self.world:
+            self.world[coord] = Point(coord, self.initial_values, self.engine)
```

Comment:
if next_vertical not in self.world:
                self.world[next_vertical] = Point(next_vertical, self.initial_values, self.engine)
                self.engine.points_changed.append(next_vertical)
                
^^^^ same code - DRY